tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -222,10 +324,144 @@ describe(__filename, () => {
         forAddonSlug: 'test',
       }));
 
       expect(getAddonsForSlug(state, 'not-a-slug')).toBeNull();
     });
   });
+
+  describe('getAddonsForUsernames selector', () => {
+    it('returns addons for a single author', () => {
+      const addons = fakeAddons();
+      const state = reducer(undefined, loadAddonsByAuthors({
+        addons: Object.values(addons),
+        authorNames: ['fakeUsername'],
+      }));
+
+      expect(getAddonsForUsernames(state, ['test2'])).toEqual([
+        createInternalAddon(addons.firstAddon),
+        createInternalAddon(addons.secondAddon),
+      ]);
+    });
+
+    it('returns addons for multiple authors of different add-ons', () => {
+      const addons = fakeAddons();
+      const state = reducer(undefined, loadAddonsByAuthors({
+        addons: Object.values(addons),
+        authorNames: ['fakeUsername'],
+      }));
+
+      expect(getAddonsForUsernames(state, ['test', 'test3'])).toEqual([
+        createInternalAddon(addons.firstAddon),
+        createInternalAddon(addons.thirdAddon),
+      ]);
+    });
+
+    it('returns addons for multiple authors that share add-ons', () => {
+      const addons = fakeAddons();
+      const state = reducer(undefined, loadAddonsByAuthors({
+        addons: Object.values(addons),
+        authorNames: ['test', 'test2'],
+      }));
+
+      expect(getAddonsForUsernames(state, ['test', 'test2'])).toEqual([
+        createInternalAddon(addons.firstAddon),
+        createInternalAddon(addons.secondAddon),
+      ]);
+    });
+
+    it('omits addon with matching slug', () => {
+      // Usually the `fetchAddonsByAuthors` action includes a `forAddonSlug`
+      // that is passed to the API as the `exclude_addons` property, meaning
+      // the same add-on won't appear in the search results of a user's
+      // other add-ons for one request, but after several requests are made
+      // we end up looking through the entire reducer, so we have this filter
+      // to prevent an add-on from appearing in its own "by this author"
+      // list.
+      const addons = fakeAddons();
+      const authorNames = ['test', 'test2', 'test3'];
+      const state = reducer(undefined, loadAddonsByAuthors({
+        addons: Object.values(addons),
+        addonType: ADDON_TYPE_EXTENSION,
+        authorNames,
+      }));
+
+      expect(getAddonsForUsernames(
+        state, authorNames, ADDON_TYPE_EXTENSION, addons.firstAddon.slug
+      )).toEqual([
+        createInternalAddon(addons.secondAddon),
+        createInternalAddon(addons.thirdAddon),
+      ]);
+    });
+
+    it('returns nothing if no add-ons are found', () => {
+      const addons = fakeAddons();
+      const state = reducer(undefined, loadAddonsByAuthors({
+        addons: Object.values(addons),
+        authorNames: ['fakeUsername'],
+      }));
+
+      expect(getAddonsForUsernames(state, ['nobody'])).toBeNull();
+    });
+  });
+
+  describe('getLoadingForAuthorNames', () => {
```

Comment:
Should we have a test for the case where `getLoadingForAuthorNames` returns `true` after a fetch, and then returns `false` after a load?